general-concepts/licenses/text.xml

```diff
@@ -81,14 +81,105 @@ of applicable licenses but the exact application of licenses and their
 versions are specified elsewhere.
 </p>
 
+</body>
+</section>
+
+<section>
+<title>Detecting and solving upstream license problems</title>
+<body>
+
 <p>
-Please watch for license conflicts. If the license indicated
-by the package is incompatible with the licenses used by its sources
-(e.g. BSD/MIT package including GPL sources), please contact
-the licenses team for guidance. Do not add packages that seem
-to include license term violations.
+While adding new packages, you need to be watchful for upstream license
+problems. Please understand that they may not only affect upstream
+but also put yourself and our users in danger. Even if you could
+reasonably assume nobody would pursue legal action against our users,
+they may be forced to put significant effort in replacing problematic
+modules.
 </p>
 
+<p>
+If you notice specific copyright problems, please report them upstream.
+You may ask the Gentoo licenses team for guidance. In general, it is
+preferable to wait for upstream to resolve the issue and release a new
+version. However, if the issue is already present in the current
+version, you may choose to apply local fixes or workarounds.
+</p>
+
+<p>
+Common license problems include but are not limited to:
+</p>
+
+<ol>
+  <li>
+    <p>
+      Including third party code without appropriate copyright notices.
+      Practically all licenses (with notable exception of public
+      domain-alike) require attribution, and some require copying
+      original copyright notices verbatim.
+    </p>
+
+    <p>
+      This issue can be fixed by readding the original copyright
+      notices. Please note that the license of the project may need
+      to be reevaluated afterwards.
+    </p>
+  </li>
+
+  <li>
+    <p>
+      Combining incompatible licenses. When you are combining multiple
+      files using different licenses into a single executable, those
+      licenses need to be compatible. For example, it is not possible
+      to combine proprietary code with copyleft licenses (e.g. GPL).
+      But it also incorrect to combine GPL-2 and GPL-3 code.
+    </p>
+
+    <p>
+      In some cases, upstream may be able to resolve the issue by
+      relicensing parts of the code to a compatible license. However,
+      this requires permission from copyright holders. If this is
+      not feasible, the incompatible code can be either removed or
+      replaced with an alternative implementation. If the original code
+      is independently rewritten, a clear room design is recommended.
+    </p>
+  </li>
+
+  <li>
+    <p>
+      Dynamically linking incompatible executables. In addition to
+      restrictions made on combining code into a single executable, some
+      licenses also apply restriction on dynamic linking between
+      executables and shared libraries. For example, normally you can't
```

Comment:
IIUC it is still an open question if licenses can restrict dynamic linking. See for example Lawrence Rosen's book (https://www.rosenlaw.com/oslbook.htm, chapter 6, page 115 ff), or his article in the Linux Journal: https://www.linuxjournal.com/article/6366

So maybe use a more careful wording, e.g. "arguably, some licenses apply restrictions ...".